philander/potentiometer.py

```diff
@@ -0,0 +1,119 @@
+        Key name                         Value type, meaning and default
+        =============================    =====================================================================================================
+        Potentiometer.resistance.max     ``int``; Maximum resistance in Ohm; :attr:`DEFAULT_RESISTANCE_MAX`.
+        Potentiometer.resolution         ``int``; Number of bits used to set resistance value.
+        ======================================================================================================================================
+        """
+        defaults = {
+            "Potentiometer.resistance.max": Potentiometer.DEFAULT_RESISTANCE_MAX,
+            "Potentiometer.resolution": Potentiometer.DEFAULT_RESOLUTION
+        }
+        defaults.update(paramDict)
+        paramDict = defaults
+        return None
+    
+    def open(self, paramDict):
+        """Initialize an instance and prepare it for use.
+
+        Also see: :meth:`.Module.open`.
+        
+        :param dict(str, object) paramDict: Configuration parameters as\
+        possibly obtained from :meth:`Params_init`.
+        :return: An error code indicating either success or the reason of failure.
+        :rtype: ErrorCode
+        """
+        return NotImplementedError
+        
+    def close(self, paramDict):
+        """Close this instance and release hardware resources.
+        
+        Also see: :meth:`.Module.close`.
+        
+        :return: An error code indicating either success or the reason of failure.
+        :rtype: ErrorCode
+        """
+        return NotImplementedError
+    
+    def _digitalize_resistance_value(value, isAbsolute=False, max_resistance=None, isDigital=False, resolution=None):
+        """Converts an either digital, absolute or relative resistance value into a digital value. It also checks for it's validity.
+
+        Also see: :meth:`.potentiometer._eval_resistance_value`.
+        
+        :param int value: Resistance value, interpreted as percentage (0 to 100) by default.\
+        Will be inpreted as absolute resistance in ohms, if isAbsolute is set true.
+        :param bool isAbsolute: Set true if value is in ohms.
+        :return: An error code indicating either success or the reason of failure.
+        :rtype: int
+        """
+        Potentiometer._eval_resistance_value(value, isAbsolute, max_resistance, isDigital, resolution)
+        pot_steps = 2 ** resolution # number of available steps, dependent on number of bits for resolution
+        if isAbsolute:
+            return round((value / max_resistance) * (pot_steps - 1))
+        elif isDigital:
+            return value
+        else:
+            return round((value / 100) * (pot_steps - 1))
+    
+    def _eval_resistance_value(value, isAbsolute=False, max_resistance=None, isDigital=False, resolution=None):

```

Comment:
Consistently document all or none of the parameters.
Avoid print(). Instead, throw an exception or return an error code.